### ISSUE TITLE
Stream.melodicIntervals() extra interval to start

### ISSUE THREAD

**Issue by asdfang** [cuthbertLab/music21#491]
I am trying to get the melodic interval sequence for a part in a Bach chorale, for example:
```
for chorale in music21.corpus.chorales.Iterator(1, 3):
  chorale.parts[0].melodicIntervals().show('txt')
  chorale.show()
```
Here I just want the melodic intervals for the top soprano line.
It seems that there is an extra interval to start – i.e. there's an extra P1 for the first two chorales and an extra P5 for the third one in the soprano line for this code.

What's happening? How can this be fixed? (there's the obvious workaround of skipping the first one if this is a known problem). Otherwise, please let me know if I'm misunderstanding something. Thanks!

Update: I see, the first interval seems to be the interval from going from the last note of the part to the first note. This seems a little unintuitive from looking at the examples from the documentation.

**Comment by mscuthbert** [comment]
I'm going to need a minimal example that reproduces this error with the fewest number of notes in order to look into it.  Thanks.

**Comment by jacobtylerwalls** [comment]
Fixed in #565, which filters out non-`GeneralNote` objects in `findConsecutiveNotes()`.

m21 was calculating the "interval" from a `Key` to the first note:
```
s = stream.Stream()
s.insert(key.Key("B"))
s.findConsecutiveNotes()
OUT: [<music21.key.Key of B major>]
```
